### DISCUSSION TITLE
Support tools

### DISCUSSION THREAD

**Discussion by switch-blade-stuff** [exodrifter/unity-python#18] [Ideas]
Ive added some support tools in my project.
For now:
- Create new script database menu option - Creates new ScriptableObject(throws a warning if created twice) that can hold all python script paths.
- Create new python script menu option - Creates new python script and adds it to database if its available.

- Database inspector menu: 
 1. Specify name of new python script
 2. Create new python script and add to database

Will these tools be useful for this project? If yes, let me know and ill polish them and send here.

**Comment by exodrifter** [comment]
I don't think your feature would fit in this repository for the following reasons:

- **Is out of scope.** This does not improve the wrapper around IronPython or make the task of integrating IronPython easier. 
- **Is not general.** While your approach may make integrating IronPython easier for you, there are many ways to manage your assets in Unity.

I would recommend making a new repository for your code if you would like to share it with others. As a side note, I'd like to point out a few things to you about your tool that you may want to consider:
- **Prone to data inconsistency.** It is difficult to automatically keep your database in sync with the actual number of python scripts in your repository. What if someone created a new python script before the database was created? What if someone already had python scripts before they imported your tool?
- **Not integrated with Unity's build approach.** Unity uses the asset links to decide what needs to be compiled into the executable. With your current approach, _all_ of the python scripts will be compiled into the executable even if the database is referenced exactly once for exactly one script. This might be a trade-off some people would be willing to have, but it's not something everyone might want.

**Comment by switch-blade-stuff** [comment]
Sure, ill consider that, thanks for responding! I made those tools for myself and i didnt have those problems, but ill think about that in case ill release those)
